tests/client_test.py

```diff
@@ -254,5 +258,60 @@ def setUp(self):
         self.uut = SwaggerClient.from_resource_listing(self.resource_listing)
 
 
-if __name__ == '__main__':
-    unittest.main()
+class MockHttpClient(object):
+
+    def __init__(self, server_response):
+        self.response = server_response
+
+    def request(self, request, response_callback):
+        return HttpFuture(
+            mock.Mock(),
+            mock.Mock(return_value=mock.Mock(
+                json=mock.Mock(return_value=self.response),
+                status_code=200)),
+            response_callback)
+
+
+@pytest.fixture
+def swagger_client():
+    spec_dict = {
+        "swagger": "2.0",
+        "info": {
+            "version": "1.0.0",
+            "title": "Simple"
+        },
+        "paths": {
+            "/pet": {
+                "get": {
+                    "operationId": "getPet",
+                    "responses": {
+                        "200": {
+                            "description": "Get a pet",
+                            "schema": {
+                                "$ref": "#/definitions/Pet",
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "definitions": {
+            "Pet": {
+                "properties": {
+                    "id": {
+                        "type": "integer",
+                    }
+                }
+            }
+        }
+    }
+    return SwaggerClient(Spec.from_dict(spec_dict, 'http://example.com/'))
+
+
+def test_client_ignores_unexpecte_response_fields(swagger_client):
+    swagger_client.swagger_spec.http_client = MockHttpClient({
+        "id": 12345,
+        "alias": "thing",
+    })
+    response = swagger_client.pet.getPet()
+    assert response.result() == dict(id=12345)
```

Comment:
Ah, that's a good point and worth adding as a comment in the test and to the README as well. As this PR is of low priority i will pick this up later.